internal/api/rolebindings.go

```diff
@@ -0,0 +1,366 @@
+
+	resourceID, err := gidx.Parse(resourceIDStr)
+	if err != nil {
+		return r.errorResponse("error parsing resource ID", fmt.Errorf("%w: %s", ErrInvalidID, err.Error()))
+	}
+
+	resource, err := r.engine.NewResourceFromID(resourceID)
+	if err != nil {
+		return r.errorResponse("error creating resource", err)
+	}
+
+	subjectResource, err := r.currentSubject(c)
+	if err != nil {
+		return err
+	}
+
+	if err := r.checkActionWithResponse(ctx, subjectResource, string(iapl.RoleBindingActionList), resource); err != nil {
+		return err
+	}
+
+	roleFilter := (*types.Resource)(nil)
+
+	if roleIDStr != "" {
+		roleID, err := gidx.Parse(roleIDStr)
+		if err != nil {
+			return r.errorResponse("error parsing role ID", fmt.Errorf("%w: %s", ErrInvalidID, err.Error()))
+		}
+
+		roleResource, err := r.engine.NewResourceFromID(roleID)
+		if err != nil {
+			return r.errorResponse("error creating role resource", err)
+		}
+
+		roleFilter = &roleResource
+	}
+
+	rbs, err := r.engine.ListRoleBindings(ctx, resource, roleFilter)
+	if err != nil {
+		return r.errorResponse("error listing role-binding", err)
+	}
+
+	resp := listRoleBindingsResponse{
+		Data: make([]roleBindingResponse, len(rbs)),
+	}
+
+	for i, rb := range rbs {
+		resp.Data[i] = roleBindingResponse{
+			ID:       rb.ID,
+			Subjects: resourceToSubject(rb.Subjects),
+			Role: roleBindingResponseRole{
+				ID:   rb.Role.ID,
+				Name: rb.Role.Name,
+			},
+		}
+	}
+
+	return c.JSON(http.StatusOK, resp)
+}
+
+func (r *Router) roleBindingsDelete(c echo.Context) error {
+	resID := c.Param("id")
+	rbID := c.Param("rb_id")
+
+	ctx, span := tracer.Start(
+		c.Request().Context(), "api.roleBindingDelete",
+		trace.WithAttributes(attribute.String("id", rbID)),
+	)
+	defer span.End()
+
+	// resource
+	resourceID, err := gidx.Parse(resID)
+	if err != nil {
+		return r.errorResponse("error parsing resource ID", fmt.Errorf("%w: %s", ErrInvalidID, err.Error()))
+	}
+
+	resource, err := r.engine.NewResourceFromID(resourceID)
+	if err != nil {
+		return r.errorResponse("error creating resource", err)
+	}
+
+	// role-binding
+	rolebindingID, err := gidx.Parse(rbID)
+	if err != nil {
+		return r.errorResponse("error parsing resource ID", fmt.Errorf("%w: %s", ErrInvalidID, err.Error()))
+	}
+
+	rbRes, err := r.engine.NewResourceFromID(rolebindingID)
+	if err != nil {
+		return r.errorResponse("error creating resource", err)
+	}
+
+	actor, err := r.currentSubject(c)
+	if err != nil {
+		return err
+	}
+
+	// permissions on role binding actions, similar to roles v1, are granted on the resources
+	if err := r.checkActionWithResponse(ctx, actor, string(iapl.RoleBindingActionDelete), resource); err != nil {
+		return err
+	}
+
+	if err := r.engine.DeleteRoleBinding(ctx, rbRes, resource); err != nil {
+		return r.errorResponse("error updating role-binding", err)
+	}
+
+	resp := deleteRoleBindingResponse{Success: true}
+
+	return c.JSON(http.StatusOK, resp)
+}
+
+func (r *Router) roleBindingGet(c echo.Context) error {
```

Comment:
As written, this doesn't actually enforce that the role ID and binding ID are related in any way. It would probably make more sense to just use the binding ID and do authZ checks accordingly.